src/Lean/Parser/Command.lean

```diff
@@ -67,7 +81,7 @@ def declModifiers (inline : Bool) := leading_parser
   optional («partial» <|> «nonrec»)
 /-- `declId` matches `foo` or `foo.{u,v}`: an identifier possibly followed by a list of universe names -/
 def declId           := leading_parser
-  ident >> optional (".{" >> sepBy1 ident ", " >> "}")
+  ident >> optional (".{" >> sepBy1 (recover ident (recoverUntil (fun c => c.isWhitespace || c ∈ [',', '}']))) ", " >> "}")
```

Comment:
I find it a bit confusing that both of these have `recover` in their name but different roles

Reply:
Good point! Fixed.